Client/Frontend/Settings/HomePageSettingsViewController.swift

```diff
@@ -0,0 +1,129 @@
+            WebPageSetting(prefs: prefs,
+                prefKey: HomePageConstants.HomePageURLPrefKey,
+                placeholder: Strings.SettingsHomePagePlaceholder,
+                accessibilityIdentifier: "HomePageSetting"),
+            ButtonSetting(title: NSAttributedString(string: Strings.SettingsHomePageUseCurrentPage),
+                accessibilityIdentifier: "UseCurrentTab",
+                isEnabled: isHomePage(currentTab),
+                onClick: setHomePage(currentTab)),
+            ButtonSetting(title: NSAttributedString(string: Strings.SettingsHomePageUseCopiedLink),
+                accessibilityIdentifier: "UseCopiedLink",
+                isEnabled: isHomePage(clipboardURL),
+                onClick: setHomePage(clipboardURL)),
+        ]
+
+        if let _ = defaultURL() {
+            basicSettings += [
+                ButtonSetting(title: NSAttributedString(string: Strings.SettingsHomePageUseDefault),
+                    accessibilityIdentifier: "UseDefault",
+                    onClick: setHomePage(defaultURL)
+                )
+            ]
+        }
+
+        basicSettings += [
+            ButtonSetting(title: NSAttributedString(string: Strings.SettingsHomePageClear),
+                destructive: true,
+                accessibilityIdentifier: "ClearHomePage",
+                onClick: setHomePage({ nil })
+            )
+        ]
+
+        var settings: [SettingSection] = [
+            SettingSection(title: NSAttributedString(string: Strings.SettingsHomePageURLSectionTitle), children: basicSettings),
+        ]
+
+        if AppConstants.MOZ_MENU {
+            settings += [
+                SettingSection(children: [
+                    BoolSetting(prefs: prefs,
+                        prefKey: HomePageConstants.HomePageButtonIsInMenuPrefKey,
+                        defaultValue: true,
+                        titleText: Strings.SettingsHomePageUIPositionTitle,
+                        statusText: Strings.SettingsHomePageUIPositionSubtitle
+                    ),
+                ]),
+            ]
+        }
+
+        return settings
+    }
+}
+
+class WebPageSetting: StringSetting {
+    init(prefs: Prefs, prefKey: String, defaultValue: String? = nil, placeholder: String? = nil, accessibilityIdentifier: String? = nil, settingDidChange: ((String?) -> Void)? = nil) {
+        super.init(prefs: prefs,
+                   prefKey: prefKey,
+                   defaultValue: defaultValue,
+                   placeholder: placeholder,
+                   accessibilityIdentifier: accessibilityIdentifier,
+                   settingIsValid: WebPageSetting.isURL,
+                   settingDidChange: settingDidChange)
+        textField.keyboardType = .URL
+        textField.autocapitalizationType = .None
+        textField.autocorrectionType = .No
+    }
+
+    static func isURL(string: String?) -> Bool {
+        return NSURL(string: string ?? "invalid://")?.isWebPage() ?? false
```

Comment:
This is kinda hacky with the `invalid://` bit. Maybe just add a `guard` on `string` above instead?